packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -259,6 +259,43 @@ void main() {
     });
   });
 
+  testWidgets('showDayOfWeek is supported only in date mode', (WidgetTester tester) async {
+    expect(
+      () => CupertinoDatePicker(
+        mode: CupertinoDatePickerMode.date,
+        onDateTimeChanged: (_) {},
+        showDayOfWeek: true,
+      ),
+      returnsNormally,
+    );
+
+    expect(
+      () => CupertinoDatePicker(
+        mode: CupertinoDatePickerMode.time,
+        onDateTimeChanged: (_) {},
+        showDayOfWeek: true,
+      ),
+      throwsAssertionError,
```

Comment:
Can we check the error message for each of these assertions, to verify the right message is given?

Reply:
I think that in some way, yes

Something like that?

```dart
expect(
    () => CupertinoDatePicker(
      mode: CupertinoDatePickerMode.monthYear,
      onDateTimeChanged: (_) {},
      showDayOfWeek: true,
    ),
    throwsA(isA<AssertionError>().having(
      (e) => e.message,
      'message',
      contains('showDayOfWeek is only supported in date mode'),
    )),
  );
```

I think that we can make some kind of util so we can pass the message to it, and call it just like the `throwsAssertionError` but, I think that we don't have it right now


I think in something like that atm:
```dart
const String _showDayOfWeekTestLabel = 'showDayOfWeek is supported only in date mode';
/// assign the name of the test to a local variable 

  testWidgets(_showDayOfWeekTestLabel, (WidgetTester tester) async {

    expect(
      () => CupertinoDatePicker(
        mode: CupertinoDatePickerMode.time,
        onDateTimeChanged: (_) {},
        showDayOfWeek: true,
      ),
    throwsA(isA<AssertionError>().having(
      (AssertionError e) => e.message,
      'message',
      contains(_showDayOfWeekLabel),
    )),
    );

```

I know...This one is ugly as hell xD, but, if we want to do it better, maybe we can add another final to `matchers.dart`.

Something like throwsAssertionErrorWithMessage<ourMessage>

Is this that we are looking for @Piinks ?

Reply:
I think having a `throwsAssertionErrorWithMessage` would be nice. For now though, I'd suggest using the first approach you mentioned because it seems very simple and straightforward.